<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <!-- Meta tags for social media banners, these should be filled in appropriatly as they are your "business card" -->
  <!-- Replace the content tag with appropriate information -->
  <meta name="description" content="PlantCaducues Project Page">
  <meta property="og:title" content="PlantCaducues"/>
  <meta property="og:description" content="PlantCaduceus: Cross-species Modeling of Plant Genomes at Single Nucleotide Resolution"/>
  <meta property="og:url" content="https://kuleshov-group.github.io/caduceus.github.io/"/>
  <!-- Path to banner image, should be in the path listed below. Optimal dimenssions are 1200X630-->
  <meta property="og:image" content="static/images/caduceus.png" />
  <meta property="og:image:width" content="630"/>
  <meta property="og:image:height" content="630"/>

  <title>PlantCaduceus project page</title>
  <style>
    .subtitle {
        margin-top: 20px; /* Adjust the value as needed */
    }
  </style>
  <link rel="icon" type="image/x-icon" href="static/images/favicon.ico">
  <link href="https://fonts.googleapis.com/css?family=Google+Sans|Noto+Sans|Castoro"
  rel="stylesheet">

  <link rel="stylesheet" href="static/css/bulma.min.css">
  <link rel="stylesheet" href="static/css/bulma-carousel.min.css">
  <link rel="stylesheet" href="static/css/bulma-slider.min.css">
  <link rel="stylesheet" href="static/css/fontawesome.all.min.css">
  <link rel="stylesheet"
  href="https://cdn.jsdelivr.net/gh/jpswalsh/academicons@1/css/academicons.min.css">
  <link rel="stylesheet" href="static/css/index.css">

  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
  <script src="https://documentcloud.adobe.com/view-sdk/main.js"></script>
  <script defer src="static/js/fontawesome.all.min.js"></script>
  <script src="static/js/bulma-carousel.min.js"></script>
  <script src="static/js/bulma-slider.min.js"></script>
  <script src="static/js/index.js"></script>
  
</head>
<body>
<section class="hero">
  <div class="hero-body">
    <div class="container is-max-desktop">
      <div class="columns is-centered">
        <div class="column has-text-centered">
          <h1 class="title is-1 publication-title">
            <img src="static/images/favicon.ico" alt="Caduceus" style="width:50px;height:50px;">
            PlantCaduceus: Cross-species Modeling of Plant Genomes at Single Nucleotide Resolution
          </h1>
          <div class="is-size-5 publication-authors">
            <!-- Paper authors -->
            <span class="author-block">
                <a href="https://www.maizegenetics.net/jingjingzhai" target="_blank">Jingjing Zhai<sup>1</sup></a>,
              </span>
            <span class="author-block">
                <a href="https://skylion007.github.io/" target="_blank">Aaron Gokaslan<sup>1</sup></a>,
            </span>
            <span class="author-block">
                <a href="https://yair-schiff.github.io/" target="_blank">Yair Schiff<sup>1</sup></a>,
            </span>
            <span class="author-block">
                <a href="https://www.maizegenetics.net/anaberthel" target="_blank">Ana Berthel<sup>1</sup></a>,
            </span>
            <span class="author-block">
                <a href="https://www.maizegenetics.net/zong-yan-liu" target="_blank">Zong-Yan Liu<sup>1</sup></a>,
            </span>
            <span class="author-block">
                <a href="https://www.maizegenetics.net/zackmiller" target="_blank">Zachary R. Miller<sup>1</sup></a>,
            </span>
            <span class="author-block">
                <a href="https://www.cshl.edu/research/postdoctoral-research/postdocs/armin-scheben/" target="_blank">Armin Scheben<sup>3</sup></a>,
            </span>
            <span class="author-block">
                <a href="https://mcstitzer.github.io" target="_blank">Michelle C. Stitzer<sup>1</sup></a>,
            </span>
            <span class="author-block">
                <a href="https://www.maizegenetics.net/cintaromay" target="_blank">M. Cinta Romay<sup>1</sup></a>,
            </span>
            <span class="author-block">
                <a href="https://www.maizegenetics.net/edwardbuckler" target="_blank">Edward S. Buckler<sup>1,2</sup></a>,
            </span>
            <span class="author-block">
                <a href="https://www.cs.cornell.edu/~kuleshov" target="_blank">Volodymyr Kuleshov<sup>1</sup></a>
            </span>
          </div>
          <div class="is-size-5 publication-authors">
            <span class="author-block">
              <sup>1</sup>
              Cornell University&nbsp;
              <img src="static/images/cornell.png" alt="Cornell" style="float:right;width:30px;height:30px;">
            </span>
            <span class="author-block">
              &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<sup>2</sup>
              USDA-ARS&nbsp;
              <img src="static/images/USDA.png" alt="USDA" style="float:right;width:40px;height:30px;">
            </span>
            <span class="author-block">
              &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<sup>3</sup>
              Cold Spring Harbor Laboratory&nbsp;
              <img src="static/images/CSHL.webp" alt="CSHL" style="float:right;width:60px;height:30px;">
              </span>
          </div>
          <div class="column has-text-centered">
            <div class="publication-links">
              <!-- ArXiv abstract Link -->
              <span class="link-block">
                <a href="https://www.biorxiv.org/content/10.1101/2024.06.04.596709v2" target="_blank" class="external-link button is-normal is-rounded is-dark">
                  <span class="icon">
                    <i class="ai ai-arxiv"></i>
                  </span>
                  <span>bioRxiv</span>
                </a>
              </span>
              <!-- Github link -->
              <span class="link-block">
                <a href="https://github.com/kuleshov-group/PlantCaduceus" target="_blank" class="external-link button is-normal is-rounded is-dark">
                  <span class="icon">
                    <i class="fab fa-github"></i>
                  </span>
                  <span>GitHub</span>
                </a>
              </span>
              <!-- HuggingFace link -->
              <span class="link-block">
                <a href="https://huggingface.co/collections/kuleshov-group/plantcaduceus-512bp-len-665a229ee098db706a55e44a" target="_blank" class="external-link button is-normal is-rounded is-dark">
                  <span class="icon">
                    <p>&#129303;</p>
                  </span>
                  <span>HuggingFace</span>
                </a>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>


<!-- Introduction -->
<section class="section" id="Introduction">
  <div class="container is-max-desktop">
    <h2 class="title">Introduction</h2>
    <div class="content is-medium">
      <p class="justified-text">
      Understanding plant genomes is crucial for plant breeding, agricultural productivity, and sustainability. However, plant genomes are diverse in size, composition, and complexity, experimentally generating analogous genomic resources for hundreds of thousands of plant species is challenging and impractical. This highlights the need for developing cross-species models capable of capturing evolutionary conservation across diverse plant species.

      </p>
    </div>
    <img src="static/images/Transfer-to-crops.png" alt="Model plant to crops" style="width:70%;height:70%; display: block; margin-left: auto; margin-right: auto;"/>
  </div>
</section>
<!-- End Introduction-->

<!-- Introduction -->
<section class="section" id="Introduction">
  <div class="container is-max-desktop">
    <h2 class="title">Why DNA Language Model?</h2>
    <div class="content is-medium">
      <p class="justified-text">
        Supervised sequence models are successful in understanding DNA sequence function but typically require large-scale labeled data, such as ENCODE-scale datasets, to achieve robust performance. Such extensive labeled data is often scarce in plant genomics. However, the success of self-supervised language models (LMs) offers a promising alternative. In this paradigm, a base model is pre-trained on vast amounts of unlabeled biological sequences to learn evolutionary conservation. Pre-trained models are then fine-tuned on limited labeled data, enabling better performance on downstream tasks and enhancing generalizability across species relative to existing methods.
      </p>
    </div>
  </div>
</section>
<!-- End Introduction-->

<!--Intro Caduceus-->
<section class="section" id="Intro PlantCaduceus">
  <div class="container is-max-desktop">
    <h2 class="title">Introducing PlantCaduceus</h2>
    <div class="content is-medium">
      <p class="justified-text">
      PlantCaduceus builds upon Caduceus <a href="https://arxiv.org/abs/2403.03234" target="_blank">(Schiff et al., 2024)</a> and Mamba <a href="https://arxiv.org/abs/2312.00752" target="_blank">(Gu et al., 2023)</a> architectures to model diverse  genomes at single nucleotide resolution. PlantCaduceus is trained on 16 Angiosperm genomes spanning 160 million years of evolutionary history, to capture evolutionary conservation and variation across species. PlantCaduceus takes 512 base pair (bp) windows of input sequences, tokenizing them into single nucleotides, and is pre-trained using a masked language modeling objective.
      </p>      
    </div>
    <div class="columns is-centered">
      <img src="static/images/PlantCaduceus.png" alt="Pretraining of PlantCaduceus" style="width:70%;height:70%; display: block; margin-left: auto; margin-right: auto;"/>
    </div>
  </div>
</section>
<!-- End Intro Caduceus--->


<!-- Experiments -->
<section class="section" id="Experiments">
  <div class="container is-max-desktop">
    <h2 class="title">Experiments</h2>
    <div>
      <h2 class="subtitle"> Benchmark models</h2>
    </div>
    <div class="content is-medium">
      <ul>
        <li>Supervised CNN + LSTM: For each task, we trained a convolutional and recurrent neural network <a href="https://academic.oup.com/nar/article/44/11/e107/2468300" target="_blank">(Quang et al., 2016)</a> from scratch using one-hot encoded DNA sequences as input. </li>
        <li>Nucleotide transformer <a href="" target="_blank">(Hugo et al., 2023)</a> is a multi-species non-plant transformer DNA LM, we used the multi-species version 2 that was pre-trained on 850 genomes, hereafter referred to as NT-v2. </li>
        <li>AgroNT <a href="https://www.biorxiv.org/content/10.1101/2023.10.24.563624v2" target="">(Javier et al., 2023)</a> is a multi-species plant transformer DNA LM pre-trained on 48 plant genomes. </li>
        <li>GPN <a href="https://www.pnas.org/doi/full/10.1073/pnas.2311219120" target="_blank">(Benegas et al., 2023)</a> is a convolutional DNA LM pre-trained on eight genomes from the Brassicales order. </li>
        </p>
    </ul>
    </div>
    <div>
      <h2 class="subtitle"> Modeling translation initiation and termination sites</h2>
    </div>

    <div class="content is-medium">
      <p class="justified-text">
        Accurate prediction of translation initiation site (TIS), translation termination site (TTS), and splice donor and acceptor sites, is crucial in gene annotation. To explore PlantCaduceus's performance on gene annotation tasks, we generated training and validation datasets for these four tasks, using Arabidopsis thaliana TAIR10, a relatively well-annotated genome released over two decades ago. To evaluate the cross-species prediction performance, we generated three extremely imbalanced cross-species testing datasets for rice, sorghum, and maize. With only training on Arabidopsis, PlantCaduceus outperformed all benchmark models on TIS and TTS tasks, outperforming the best benchmark model GPN by 7.23-fold and 3.75-fold for TIS and TTS when transfered to 160 million year diverged maize, respectively.
      </p>
    </div>
    <div class="columns is-centered">
      <div class="column is-four-fifths">
        <img src="static/images/experiments/Modeling_translation.png" alt="Modeling translation" style="width:100%;height:100%; display: block; margin-left: auto; margin-right: auto;">
      </div>
    </div>
    
    <div>
      <h2 class="subtitle">Modeling splice donor and acceptor sites</h2>
    </div>

    <div class="content is-medium">
      <p class="justified-text">
        With similar way as TIS and TTS, we generated training, validation and testing datasets for splice donor and acceptor tasks. We found PlantCaduceus outperformed all benchmark models on splice donor and acceptor sites tasks, outperforming the best benchmark model GPN by 1.47-fold and 1.45-fold for splice donor and acceptor sites, respectively.
      </p>
    </div>
    <div class="columns is-centered">
      <div class="column is-four-fifths">
        <img src="static/images/experiments/Modeling_transcription.png" alt="Modeling transcription" style="width:100%;height:100%; display: block; margin-left: auto; margin-right: auto;">
      </div>
    </div>

    <div>
      <h2 class="subtitle">Prioritizing deleterious mutations by zero-shot strategy</h2>
    </div>

    <div class="content is-medium">
      <p class="justified-text">
        The training objective of PlantCaduceus is to predict masked nucleotides based on sequence context; if a pre-trained multi-species DNA LM can accurately predict masked tokens, it suggests that similar sequence patterns, conserved across different species, were frequently observed during pre-training. We hypothesize that the predicted likelihood of the reference allele versus the alternate allele can identify deleterious mutations, as mutations in conserved regions across species are likely deleterious. 
      </p>
      <p class="justified-text"></p>
        Deleterious mutations tend to have lower frequencies within a population due to selective constraints, we therefore used minor allele frequency (MAF) to quantify the deleteriousness of mutations predicted by different methods. Despite the potential for low MAF in neutral/beneficial alleles, we believe this approach provides useful signals for assessing deleterious mutations. Both phyloP and phastCons assess evolutionary constraint using multiple sequence alignments and phylogenetic models, assigning higher scores to conserved regions. We found the deleterious mutations identified with the zero-shot strategy of PlantCaduceus show a three-fold enrichment of rare alleles compared to phastCons. For missense mutations, PlantCaduceus matches the performance of the state-of-the-art protein LM.
      </p>
    </div>
    <div class="columns is-centered">
      <div class="column is-four-fifths">
        <img src="static/images/experiments/Deleterious_mutations.png" alt="Deleterious mutations" style="width:100%;height:100%; display: block; margin-left: auto; margin-right: auto;">
      </div>
    </div>
    </div>


  <!-- End Experiments-->

<!-- Conclusion -->
<section class="section" id="Conclusion">
  <div class="container is-max-desktop">
    <h2 class="title">Conclusion</h2>
    <div class="content is-medium">
      <p class="justified-text">
        We introduced PlantCaduceus, a multi-species plant DNA LM pretrained on a curated set of 16 evolutionarily distant Angiosperm genomes, enabling cross-species prediction of functional annotations with limited data. PlantCaduceus leverages Mamba and Caduceus architectures to support bi-directional, reverse complement equivariant sequence modeling. We demonstrated the superior cross-species performance of PlantCaduceus on five tasks involving transcription, translation, and evolutionary constraint modeling. These results highlight the potential of PlantCaduceus to serve as a foundational model for comprehensively understanding plant genomes.
      </p>
    </div>
  </div>
<!-- End Experiments-->

<!--BibTex citation -->
  <section class="section" id="BibTeX">
    <div class="container is-max-desktop content">
      <h2 class="title">BibTeX</h2>
      <pre><code>
        @article {Zhai2024.06.04.596709,
          author = {Zhai, Jingjing and Gokaslan, Aaron and Schiff, Yair and Berthel, Ana and Liu, Zong-Yan and Miller, Zachary R and Scheben, Armin and Stitzer, Michelle C and Romay, Cinta and Buckler, Edward S. and Kuleshov, Volodymyr},
          title = {Cross-species plant genomes modeling at single nucleotide resolution using a pre-trained DNA language model},
          elocation-id = {2024.06.04.596709},
          year = {2024},
          doi = {10.1101/2024.06.04.596709},
          URL = {https://www.biorxiv.org/content/early/2024/06/05/2024.06.04.596709},
          eprint = {https://www.biorxiv.org/content/early/2024/06/05/2024.06.04.596709.full.pdf},
          journal = {bioRxiv}
      }      
      </code></pre>
    </div>
</section>
<!--End BibTex citation -->


  <footer class="footer">
  <div class="container">
    <div class="columns is-centered">
      <div class="column is-8">
        <div class="content is-medium">

          <p>
            This page was built using the <a href="https://github.com/eliahuhorwitz/Academic-project-page-template" target="_blank">Academic Project Page Template</a> which was adopted from the <a href="https://nerfies.github.io" target="_blank">Nerfies</a> project page.
            <br> This website is licensed under a <a rel="license"  href="http://creativecommons.org/licenses/by-sa/4.0/" target="_blank">Creative
            Commons Attribution-ShareAlike 4.0 International License</a>.
          </p>

        </div>
      </div>
    </div>
  </div>
</footer>

<!-- Statcounter tracking code -->

  <!-- You can add a tracker to track page visits by creating an account at statcounter.com -->

<!-- End of Statcounter Code -->

</body>
</html>
